torch_xla/csrc/tensor_util.cpp

```diff
@@ -664,7 +687,14 @@ torch::lazy::BackendDataPtr TensorToXlaData(
 
   static const bool transfer_async =
       xla::sys_util::GetEnvBool("XLA_TRANSFER_SCALAR_ASYNC", false);
-  if (transfer_async && tensor.dim() == 0 && tensor.numel() == 1) {
+  if (ShardingUtil::UseVirtualDevice()) {
```

Comment:
Then should we move this above `if (device.type() == (int8_t)XlaDeviceType::SPMD)`:
```
if (ShardingUtil::UseVirtualDevice()) {
if (device.type() == (int8_t)XlaDeviceType::SPMD) {
...
}
// The tensor is bypassing the virtual device, so it should be replicated
// to all devices.
...
}
```

Reply:
Sure, I'll do that now